tests/scripts/all.sh

```diff
@@ -637,6 +666,8 @@ component_test_large_ecdsa_key_signature () {
 component_test_default_out_of_box () {
     msg "build: make, default config (out-of-box)" # ~1min
     make
+    # Disable fancy stuff
+    unset MBEDTLS_TEST_OUTCOME_FILE
```

Comment:
Won't this affect all subsequent tests?

Reply:
Oh yes it will. I went

1. Check that each test case is run in a subprocess.
2. No, that's not the case. We should change that some day and now is a good time as any.
3. ...
4. Now that each test case is run in a subprocess, I can fiddle with the environment.

I'll add that missing step 3.

Reply:
On second thoughts, no, I'm not going to do step 3. It requires more changes than I want to do here. `record_status` needs to run in the main process to update the failure information. I do have it in my long-term plans to simplify this mechanism (which dates back from before `all.sh` was split into component functions), but it's out of scope here. Instead I'll just save and restore `MBEDTLS_TEST_OUTCOME_FILE`.